azure/services/managedclusters/managedclusters.go

```diff
@@ -184,240 +70,57 @@ func (s *Service) Reconcile(ctx context.Context) error {
-		}
-
-		// Avoid changing agent pool profiles through AMCP and just use the existing agent pool profiles
-		// AgentPool changes are managed through AMMP
-		managedCluster.AgentPoolProfiles = existingMC.AgentPoolProfiles
-
-		diff := computeDiffOfNormalizedClusters(managedCluster, existingMC)
-		if diff != "" {
-			klog.V(2).Infof("Update required (+new -old):\n%s", diff)
-			result, err = s.Client.CreateOrUpdate(ctx, managedClusterSpec.ResourceGroupName, managedClusterSpec.Name, managedCluster, customHeaders)
-			if err != nil {
-				return fmt.Errorf("failed to update managed cluster, %w", err)
-			}
-		}
-	}
-
-	// Update control plane endpoint.
-	if result.ManagedClusterProperties != nil && result.ManagedClusterProperties.Fqdn != nil {
-		endpoint := clusterv1.APIEndpoint{
-			Host: *result.ManagedClusterProperties.Fqdn,
-			Port: 443,
+			return errors.Wrap(err, "failed to get credentials for managed cluster")
 		}
-		s.Scope.SetControlPlaneEndpoint(endpoint)
-	} else {
-		// Fail if cluster api endpoint is not available.
-		return fmt.Errorf("failed to get API endpoint for managed cluster")
-	}
-
-	// Update kubeconfig data
-	// Always fetch credentials in case of rotation
-	kubeConfigData, err := s.Client.GetCredentials(ctx, s.Scope.ResourceGroup(), s.Scope.ClusterName())
-	if err != nil {
-		return errors.Wrap(err, "failed to get credentials for managed cluster")
-	}
-	s.Scope.SetKubeConfigData(kubeConfigData)
-
-	return nil
-}
-
-func handleAddonProfiles(managedCluster containerservice.ManagedCluster, spec azure.ManagedClusterSpec) {
-	for i := range spec.AddonProfiles {
-		if managedCluster.AddonProfiles == nil {
-			managedCluster.AddonProfiles = map[string]*containerservice.ManagedClusterAddonProfile{}
-		}
-		item := spec.AddonProfiles[i]
-		addonProfile := &containerservice.ManagedClusterAddonProfile{
-			Enabled: &item.Enabled,
-		}
-		if item.Config != nil {
-			addonProfile.Config = *to.StringMapPtr(item.Config)
-		}
-		managedCluster.AddonProfiles[item.Name] = addonProfile
+		s.Scope.SetKubeConfigData(kubeConfigData)
 	}
+	s.Scope.UpdatePutStatus(infrav1.ManagedClusterRunningCondition, serviceName, resultErr)
+	return resultErr
 }
 
 // Delete deletes the managed cluster.
 func (s *Service) Delete(ctx context.Context) error {
 	ctx, _, done := tele.StartSpanWithLogger(ctx, "managedclusters.Service.Delete")
 	defer done()
 
-	klog.V(2).Infof("Deleting managed cluster  %s ", s.Scope.ClusterName())
-	err := s.Client.Delete(ctx, s.Scope.ResourceGroup(), s.Scope.ClusterName())
-	if err != nil {
-		if azure.ResourceNotFound(err) {
-			// already deleted
-			return nil
-		}
-		return errors.Wrapf(err, "failed to delete managed cluster %s in resource group %s", s.Scope.ClusterName(), s.Scope.ResourceGroup())
+	ctx, cancel := context.WithTimeout(ctx, reconciler.DefaultAzureServiceReconcileTimeout)
+	defer cancel()
+
+	managedClusterSpec := s.Scope.ManagedClusterSpec(ctx)
+	if managedClusterSpec == nil {
```

Comment:
Maybe add a log statement similar to the other async services?

Reply:
don't think we're logging anything for other services if there is no spec? It just means that service is not applicable for that object